### DISCUSSION TITLE
24/7 Recording but events only in zone

### DISCUSSION THREAD

**Discussion by gettingerr** [blakeblackshear/frigate#3467] [Ask A Question] [answered]
Good evening. I am looking to keep 24/7 recording enabled but only get event tags when a car pulls into my driveway. I have driveway zones enabled and have in the camera config:
`    record:
      events:
        max_seconds: 300
        pre_capture: 5
        post_capture: 5
        # Optional: Restrict recordings to objects that entered any of the listed zones (default: no required zones)
        required_zones: [Driveway_Near,Driveway_Mid,Driveway_Far]`

With this, I get 24/7 recording but I also get events tagged when cars drive in the road. If I disable 24/7 Record by changing the retain to 0, I do not get events tagged for cars driving in the road. What am I doing wrong here? Or is this not possible?

**Comment by NickM-27** [comment, marked as answer]
Do you have `requried_zones` set for snapshots as well? You need it for both to avoid events for objects that do not enter a zone

ex:

```yaml
record:
  events:
    required_zones: ...

snapshots:
  required_zones: ...
```

**Reply by gettingerr** [reply]
You're a hero

**Reply by gettingerr** [reply]
And I am an absolute monkey

**Reply by NickM-27** [reply]
It's not exactly intuitive, glad that it was an easy fix

**Reply by simona70** [reply]
i use events: and required_zones: inside record: but ii is not accepted....

**Reply by NickM-27** [reply]
@simona70 share your formatted config. If you follow my code snippet above (with same spacing) it will work

**Comment by akcasoy** [comment]
I do not think that this answer is still valid.. In the full reference page https://docs.frigate.video/configuration/reference/, there is not a "required_zones" attribute anymore other than snapshot, review/alerts, review/detections

**Reply by NickM-27** [reply]
in this case `alerts -> required_zones` is the field
